src/components/i-calendar.vue

```diff
@@ -0,0 +1,537 @@
+            <button
+              class="btn-month"
+              :class="{ selected: isSelectedMonth(month.monthValue) }"
+              @click="clickMonth(month)"
+            >
+              <div class="btn-month--label">
+                {{ month.monthDisplay }}
+              </div>
+              <div
+                v-if="isThereNewUpdate"
+                class="new-update-indicator"
+              />
+            </button>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="i-calendar--body">
+      <div class="i-calendar--list-day-wrapper">
+        <div
+          v-for="day in listDays"
+          :key="day"
+          class="list-day"
+        >
+          {{ day }}
+        </div>
+      </div>
+      <div
+        ref="refCurrentPick"
+        class="i-calendar--date-list"
+      >
+        <template v-if="previousPicker.length < 7">
+          <div
+            v-for="date in previousPicker"
+            :key="`${date.date()}${date.month()}${date.year()}-previous`"
+            class="each-date previous-date"
+            :data-tailwind-calendar="date.date()"
+          >
+            <div class="hidden-date">
+              {{ date.date() }}
+            </div>
+          </div>
+        </template>
+        <div
+          v-for="(date, index) in currentPicker"
+          :key="`${date.date.date()}${date.date.month()}${date.date.year()}-current`"
+          class="each-date current-date"
+        >
+          <button
+            class="i-calendar-pick"
+            :class="{
+              selected: isSelectedDate(date.date),
+            }"
+            @click="clickDate(date.date)"
+          >
+            <div
+              class="date-number"
+              :class="{
+                red: date.date.day() === 0,
+                selected: isSelectedDate(date.date),
+              }"
+            >
+              <span>{{ date.date.date() }}</span>
+            </div>
+            <div class="content-container">
+              <slot
+                name="content"
+                :is-selected="isSelectedDate(date.date)"
+                :date="date.date"
+                :index="index"
+              />
+            </div>
+          </button>
+        </div>
+        <div
+          v-for="date in nextPicker"
+          :key="`${date.date()}${date.month()}${date.year()}-next`"
+          class="each-date next-date"
+        >
+          <div class="hidden-date">
+            {{ date.date() }}
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import dayjs from 'dayjs';
+
+import IcAngleCircle from '@/icons/ic-angle-circle.vue';
+
+export default {
+  name: 'ICalendar',
+  components: {
+    IcAngleCircle,
+  },
+  props: {
+    value: {
+      type: [Date, Array],
+      default: () => null,
+    },
+    initialDate: {
+      type: Date,
+      default: undefined,
+    },
+    disabledPreviousMonth: Boolean,
+    disabledNextMonth: Boolean,
```

Comment:
disabledPreviousMonth and disabledNextMonth should be not a prop..
if these variables was copied from datepicker, you can use again with adjustment in it function.

Reply:
ok